### ISSUE TITLE
[CLI]: MlflowImporter can NOT import metric historys whose length are different

### ISSUE THREAD

**Issue by ykihong0** [wandb/wandb#5754]
### Describe the bug

In [MlflowRun] class, it is assumed that all metric history has same length.
The reason why I think is MlflowRun use`metrics = zip(*metrics)` in metrics() (https://github.com/wandb/wandb/blob/c5fe3d56b155655980611d32ef09df35cd336872/wandb/apis/importers/mlflow.py#L45)

This assumption is not general and not compatible when mlflow run has metrics whose lengths are different. For example, MlflowCallback in huggingface transformers (https://huggingface.co/docs/transformers/v4.30.0/en/main_classes/callback#transformers.integrations.MLflowCallback)

```python
def importDataFromMlflow(
  wandb_entity:str="",
  wandb_project:str="",
  wandb_api_key:str="",
  mlflow_run_id:str=None,
  mlflow_tracking_uri:str="",
):
  from wandb.apis.importers import MlflowImporter
  import wandb
  wandb.login(key=wandb_api_key)
  
  # optional dict to override settings for all imported runs
  overrides = {
      "entity": wandb_entity,
      "project": wandb_project
  }

  importer = MlflowImporter(mlflow_tracking_uri=mlflow_tracking_uri)
  # importer.import_all_parallel()

  for run in importer.download_all_runs():
    if mlflow_run_id and run.run_id() != mlflow_run_id:
      continue
```

```shell
It is not syntax error. There is no exception traceback
```

### In Mlflow
One run has below metrics
1. epoch : [(step=10, value=0.1), (step=20, value=0.2), (step=30, value=0.3)]
2. learning_rate : [(step=10, value=0.001), (step=20, value=0.0011), (step=30, value=0.0012)]
3. eval_benchmark_acc : [(step=30, value=0.85)]

### In Wandb after MlflowImporter
Imported Wandb run has below metrics
1. epoch : [(step=1, value=0.1)]
2. learning_rate : [(step=1, value=0.001)]
3. eval_benchmark_acc : [(step=1, value=0.85)]

### Environment

WandB version: 0.15.4

OS: macOS Ventura 13.2.1

Python version: 3.8.12

Versions of relevant libraries: mlflow==2.3.0

**Comment by rsanandres-wandb** [comment]
Hello @ykihong0 ! Thanks for reporting this and I believe that this is related to the bug that you reported in this other [GitHub bug post](https://github.com/wandb/wandb/issues/5755). I will be making a bug report for this, thanks again!
